### ISSUE TITLE
seccomp: architecture is not present for (at least) ppc64le

### ISSUE THREAD

**Issue by cyphar** [moby/moby#48471]
### Description

We hit this with runc earlier this year (https://github.com/opencontainers/runc/pull/4219), but I forgot to open an issue here. Basically, for some architectures Docker didn't include the native architecture in the set of architectures for the seccomp filter, which lead to the `-ENOSYS` stub to not work.

It seems like the runtime architecture should always be set by Docker, but we have fixed the issue in runc already.

https://bugzilla.suse.com/show_bug.cgi?id=1192051#c6 is some more debugging info.

### Reproduce

1. Without https://github.com/opencontainers/runc/pull/4219, run a Docker container on ppc64le.
2. Try to call a syscall with a really large syscall number.
3. Note that the `-ENOSYS` stub doesn't work and you get `-EPERM`.

### Expected behavior

Docker should set the architecture for seccomp so the `-ENOSYS` stub is always filled properly even on older runc versions.

### docker version

```bash
n/a
```

### docker info

```bash
n/a
```

**Comment by thaJeztah** [comment]
Sorry for the late reply; saw the ticket but it dropped of my list.

I'm slightly curious though, as I see mention of "non-compliant" in the runc ticket, and wondering if the spec (which was based on the docker implementation?) is incorrect if this is not considered valid. I had a quick look at the original implementation, and that seems to indicate the list to be _optionally_ set; https://github.com/moby/moby/blob/6707f4b9b638b367a1dde6f8684a5b9817a882f0/daemon/execdriver/native/seccomp.go#L42-L54

**Comment by thaJeztah** [comment]
Oh; I may be wrong there; the original PR refers to the spec, but links is broken, so will have to check; https://github.com/moby/moby/pull/17989

> based off the runc runtime spec https://github.com/opencontainers/specs/blob/master/runtime-config-linux.md#seccomp

**Comment by cyphar** [comment]
Yeah, it is actually optional but the spec doesn't really say how we should interpret that (the libseccomp semantics are that you get the native architecture by default, which is what we now do for the `ENOSYS` stub).

**Comment by grooverdan** [comment]
Note this is more than just theoretical, hit this with Docker Engine < 26.0.0 (aka 6f242f1a28a9e658a6dd29a314cd2d1923427579 which allowed `fchmodat2`), 6.1.0-22-powerpc64le kernel running a Debian sid container:

```
## Recursive chmod
buildbot@8b18f42d0932:~$ strace chmod -R u+rw tmp
execve("/usr/bin/chmod", ["chmod", "-R", "u+rw", "tmp"], 0x7fffec795c78 /* 11 vars */) = 0
...
fchmodat2(4, "file", 0644, AT_SYMLINK_NOFOLLOW) = -1 EPERM (Operation not permitted)
fcntl(1, F_GETFL)                       = 0x10002 (flags O_RDWR|O_LARGEFILE)
write(2, "chmod: ", 7chmod: )                  = 7
write(2, "changing permissions of 'tmp/fil"..., 34changing permissions of 'tmp/file') = 34
write(2, ": Operation not permitted", 25: Operation not permitted) = 25
..
exit_group(1)                           = ?
+++ exited with 1 +++
```

Other arches corrected returned `ENOSYS` which allowed a fallback to the alternate `fchmod` as implemented in glibc.

Original problem being solved ref: https://buildbot.mariadb.org/#/builders/202/builds/21042/steps/4/logs/stdio
```
	cp --reflink=auto -a ./debian/additions/innotop/changelog.innotop debian/mariadb-client/usr/share/doc/mariadb-client
	chmod -R u\+rw,go=rX debian/mariadb-client/usr/share/doc
chmod: changing permissions of 'debian/mariadb-client/usr/share/doc/mariadb-client': Operation not permitted
chmod: changing permissions of 'debian/mariadb-client/usr/share/doc/mariadb-client/README.md': Operation not permitted
chmod: changing permissions of 'debian/mariadb-client/usr/share/doc/mariadb-client/changelog.innotop': Operation not permitted
dh_installdocs: error: chmod -R u\+rw,go=rX debian/mariadb-client/usr/share/doc returned exit code 1
make: *** [debian/rules:211: binary] Error 25
dpkg-buildpackage: error: debian/rules binary subprocess returned exit status 2
```

**Comment by cyphar** [comment]
@grooverdan We worked around this in runc 1.2. (If you're not using runc 1.2.z, I would highly suggest it because runc 1.1.z and earlier are completely unsupported now and runc 1.2.z will only get very few backports now that runc 1.3.z has been released.)
